src/bp/ui-admin/src/telemetry.ts

```diff
@@ -0,0 +1,219 @@
+  return !_.includes(info, '')
+}
+
+export async function sendServerPackage() {
+  try {
+    const {
+      data: { events }
+    } = await api.getSecured().get(serverUrl)
+    console.log(events)
+
+    const feedback = { events: [], status: '' }
+
+    if (events) {
+      events.map((obj) => (obj.source = 'client'))
+      console.log(events)
+
+      feedback.events = events.map((obj) => obj.uuid)
+
+      try {
+        await axios.post(endpoint, events, corsConfig)
+        feedback['status'] = 'ok'
+      } catch (err) {
+        feedback['status'] = 'fail'
+        console.log('Could not send the telemetry packages to the storage server', err)
+      }
+    }
+
+    await api.getSecured().post(serverUrl, feedback)
+  } catch (err) {
+    console.log('Could not access the botpress server', err)
+  }
+}
+
+export function setupServerPackageLoop() {
+  sendServerPackage().catch()
+  setInterval(async () => await sendServerPackage(), ms('1h'))
+}
+
+export async function startTelemetry(event_type: string, data: dataType, name: string = 'data') {
+  await sendTelemetry(getTelemetryPackage(event_type, data, name), event_type)
+}
+
+export function setupTelemetry() {
+  setupEventsType()
+
+  store.subscribe(() => {
+    const state = store.getState()
+
+    // @ts-ignore
+    if (_.has(state, 'version.currentVersion') && state.version.currentVersion !== '') {
+      // @ts-ignore
+      info.bp_release = state.version.currentVersion
+    }
+
+    if (_.has(state, 'license.licensing.isPro')) {
+      // @ts-ignore
+      info.bp_license = state.license.licensing.isPro ? 'pro' : 'community'
+    }
+
+    if (_.has(state, 'user.profile.email')) {
+      // @ts-ignore
+      info.email = state.user.profile.email
+    }
+
+    if (checkInfoReceived()) {
+      eventsType.forEach((event) => {
+        if (!locks[event]) {
+          changeLock(event)
+          const data = {
+            user: {
+              email: toHash(info.email),
+              timezone: Intl.DateTimeFormat().resolvedOptions().timeZone
+            },
+            language: shared.lang.getLocale()
+          }
+
+          startTelemetry(event, data).catch((err) => {
+            console.log(err)
+          })
+        }
+      })
+    }
+  })
+}
+
+function getDataCluster(data: dataType): EventData {
+  const baseCluster: EventData = {
+    schema: dataClusterVersion
+  }
+  return _.assign(baseCluster, data)
+}
+
+export function getTelemetryPackage(event_type: string, data: dataType, name: string = 'data'): TelemetryPackage {
+  return {
+    schema: telemetryPackageVersion,
+    uuid: uuid.v4(),
+    timestamp: new Date().toISOString(),
+    bp_release: info.bp_release,
+    bp_license: info.bp_license,
+    event_type: event_type,
+    source: 'client',
+    event_data: getDataCluster(data)
+  }
+}
+
+async function sendTelemetry(data: TelemetryPackage, event: string) {
+  const res = await axios.post(endpoint, data, corsConfig)
+  addTimeout(event, ms('8h'))
```

Comment:
res is not used. Could merge with the previous method.

Also: Why are every single method exported from the file ? Unless they are required outside your file elsewhere, don't export them